### ISSUE TITLE
Add speaker name to iCal/ICS event description

### ISSUE THREAD

**Issue by pferreir** [indico/indico#2069]
This piece of information is currently now exported. It would be nice to have it.

**Comment by ThiefMaster** [comment]
i.e. the chairpersons? or nothing at all for meetings/conferences?

**Comment by pferreir** [comment]
Yeah, in meetings that would be the chairperson. I guess the same goes for conferences, even though it's not that important.

**Comment by foice** [comment]
I think as well this is a very important information missing.

**Comment by foice** [comment]
I know this has been fixed and newer releases blend in the Speaker information in the Description field. However, I was wondering if a bigger set of fields than what is currently exported could be added to the iCal output. As far as I know ics files allow to have non-standard fields and there is even a [best practice suggested to handle them.](https://icalendar.org/New-Properties-for-iCalendar-RFC-7986/3-backwards-compatible-extension-properties.html)
If such non-standard fields could be produced by Indico it would be easy to import Indico events in events aggregators, look for instance here https://github.com/roed314/seminars/issues/19.

**Comment by pferreir** [comment]
> If such non-standard fields could be produced by Indico it would be easy to import Indico events in events aggregators

Why not use the [HTTP API](https://docs.getindico.io/en/stable/http_api/) in that case?

**Comment by foice** [comment]
This sounds pretty interesting! But am I correct when I say that even for
retrieving just public info an API key is mandatory?
I am asking  because from the docs it is not clear that the API key can be
omitted.

The reason why this is relevant is that I am making an "aggregator" of
events that appear in different indico categories and possibly different
indico servers. If an API key is mandatory then I need to create a user for
each indico server and make a key for each of them, plus handling the
correct key for each request. This appears as overhead if I need to just
retrieve public info.

Do you see a way out of this?

Thanks for your help.

**Comment by ThiefMaster** [comment]
Currently it's up to the Indico server's settings if and how API keys are required for accessing public content.

But for any instance where you can create an account, you can also create an API key from the account settings page.

**Comment by foice** [comment]
Thanks for this clarification! Will try it out on the servers I have need to access and see what happens.
